### ISSUE TITLE
disabling "debug" log messages?

### ISSUE THREAD

**Issue by soakes** [st3fan/dovecot-xaps-plugin#6]
Hi Guys,

I have been trying to figure out away to stop the unnecessary debug log messages which this plugin seems to produce.  I have gone though the code but I can't seem to figure out how to disable the debug messages and they are filling up my logs quickly.

Does anyone know how I can disable these messages? or do I need to "hack" the source code somewhere to do this? if so where?

Kind Regards,

Simon

#### /var/log/mail.log

```
Feb  9 22:55:57 mail1 dovecot: lda(simon@example.com): Debug: xaps_plugin_init
Feb  9 22:55:57 mail1 dovecot: lda(simon@example.com): Debug: xaps_transaction_begin
Feb  9 22:55:57 mail1 dovecot: lda(simon@example.com): Debug: xaps_transaction_begin
Feb  9 22:55:57 mail1 dovecot: lda(simon@example.com): Debug: xaps_save_finish
Feb  9 22:55:57 mail1 dovecot: lda(simon@example.com): Debug: xaps_transaction_commit
Feb  9 22:55:57 mail1 dovecot: lda(simon@example.com): sieve: msgid=<20150209225555.B486386468@mail1.example.com>: stored mail into mailbox 'INBOX'
Feb  9 22:55:57 mail1 dovecot: lda(simon@example.com): Debug: xaps_plugin_deinit
```

**Comment by soakes** [comment]
I have just found a temp solution after re-reading the source code.

Until the real fix is known from the developer or anyone who knows C, your welcome to make the same changes as I have, which will remove the debug log lines.

In short, all I have done is commented out the debug lines and then recompiled the plugin.  I expect there is a much cleaner way to deal with them i.e a debug flag in the plugin config file but alais I coulnt find it, so this is my "temp" fix.

#### Workaround:

```
--- xaps-plugin.c       2015-02-09 23:55:39.873881674 +0000
+++ xaps-plugin.c.org   2015-02-09 23:56:18.053881378 +0000
@@ -142,7 +142,7 @@
 
 static struct mailbox_transaction_context *xaps_transaction_begin(struct mailbox *box, enum mailbox_transaction_flags flags)
 {
-  //i_debug("xaps_transaction_begin");
+  i_debug("xaps_transaction_begin");
 
   struct xaps_mailbox *xaps_mailbox = XAPS_CONTEXT(box);
   xaps_mailbox->message_count = 0;
@@ -154,7 +154,7 @@
 
 static int xaps_save_finish(struct mail_save_context *ctx)
 {
-  //i_debug("xaps_save_finish");
+  i_debug("xaps_save_finish");
 
   struct mailbox_transaction_context *t = ctx->transaction;
   struct xaps_mailbox *xaps_mailbox = XAPS_CONTEXT(t->box);
@@ -170,7 +170,7 @@
 
 static int xaps_transaction_commit(struct mailbox_transaction_context *t, struct mail_transaction_commit_changes *changes_r)
 {
-  //i_debug("xaps_transaction_commit");
+  i_debug("xaps_transaction_commit");
 
   /*
    * If the message count in this transaction is not zero then we have
@@ -219,12 +219,12 @@
 
 void xaps_plugin_init(struct module *module)
 {
-  //i_debug("xaps_plugin_init");
+  i_debug("xaps_plugin_init");
   mail_storage_hooks_add(module, &xaps_mail_storage_hooks);
 }
 
 void xaps_plugin_deinit(void)
 {
-  //i_debug("xaps_plugin_deinit");
+  i_debug("xaps_plugin_deinit");
   mail_storage_hooks_remove(&xaps_mail_storage_hooks);
 }
```

**Comment by st3fan** [comment]
Fill fix this as part of issue #13. Closing this one for now because I'll probably do something a little different.
